### PR TITLE
refactor(tools): separate HTTP-related functions from tool.utils

### PR DESCRIPTION
**Pull request by Water-Melon**
### Summary

Move HTTP-related functions from `kong/tools/utils.lua` to `kong/tools/http.lua`.

### Issue reference

KAG-2957

### CODE REVIEW

File: kong/tools/http.lua [hunk 1]
```diff
@@ -0,0 +1,587 @@
...
+end
+
+
+--- Validates a cookie name.
+-- @param name (string) the cookie name to verify
+-- @return the valid cookie name, or `nil+error`
+_M.validate_cookie_name = _M.validate_http_token
+
+
+local validate_labels
```
Comment (chronolaw) [inline kong/tools/http.lua:278]: It seems that `validate_labels` is used by conf_loader, it does not belong to http.

Comment (Water-Melon) [reply kong/tools/http.lua]: Yes, it seems I overlooked this. I will create a new ticket to address it. The code has been moved back to `kong/tools/utils`.

File: kong/tools/utils.lua [hunk 2]
```diff
@@ -291,6 +30,7 @@ do
   local KEY_PTRN  = "(?!kong)(?!konnect)(?!insomnia)(?!mesh)(?!kic)" .. BASE_PTRN
   local VAL_PTRN  = BASE_PTRN
 
+
```
Comment (chronolaw) [inline kong/tools/utils.lua:33]: should not add this blank line, we should keep the original code.

File: kong/tools/utils.lua [hunk 3]
```diff
@@ -302,6 +42,7 @@ do
     return true
   end
 
+
```
Comment (chronolaw) [inline kong/tools/utils.lua:45]: same as above.

File: kong/tools/utils.lua (hunk 2 above)
Comment (Water-Melon) [reply kong/tools/utils.lua]: done

File: kong/tools/utils.lua (hunk 3 above)
Comment (Water-Melon) [reply kong/tools/utils.lua]: done

**Review by chronolaw — APPROVED** [review]

**Review by chobits — APPROVED** [review]

**Review by ADD-SP — APPROVED** [review]

**Pull request merged**